app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+class PostsController < ApplicationController
+  before_action :check_post, only: %i[edit update destroy]
+  before_action :current_user_post, only: %i[edit update destroy]
+  before_action :user_signed_in?, only: %i[create destroy edit update]
+
+  def index
+    @posts = Post.all.order('created_at DESC')
+  end
+
+  def new
+    @post = current_user.posts.build
+  end
+
+  def create
+    @post = current_user.posts.build(post_params)
+    if @post.save
+      flash[:success] = 'Post created successful!'
+    else
+      flash[:danger] = 'Unable to create your post'
+    end
+    redirect_to root_path
+  end
+
+  def edit
+    @my_post = Post.find(params[:id])
+  end
+
+  def update
+    @post = Post.find(params[:id])
+    if @post.update(post_params)
+      flash[:success] = 'Successful updated post'
+      redirect_to root_path
+    else
+      flash[:warning] = 'Unable to update post, please try again'
+      render 'edit'
+    end
+  end
+
+  def destroy
+    @post.destroy
+    flash[:success] = 'Post successful deleted'
+    redirect_to root_path
+  end
```

Comment:
- [ ] When I tried to delete a created post I got this error: 
![Screenshot from 2019-12-06 11-46-45](https://user-images.githubusercontent.com/30977845/70331380-1c3bfb80-181e-11ea-8c3a-8484b191a11f.png)